scene/main/viewport.cpp

```diff
@@ -3889,7 +3888,7 @@ void Viewport::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_vrs_texture"), &Viewport::get_vrs_texture);
 
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "disable_3d"), "set_disable_3d", "is_3d_disabled");
-	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "use_xr"), "set_use_xr", "is_using_xr");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "viewport_mode", PROPERTY_HINT_ENUM, "2D_AND_3D,3D,XR"), "set_viewport_mode", "get_viewport_mode");
```

Comment:
Here is where I mean we shouldn't remove `ADD_PROPERTY(PropertyInfo(Variant::BOOL, "use_xr"), "set_use_xr", "is_using_xr");`

We should keep it but have `set_use_xr` and `is_using_xr` be based on the value of `viewport_mode`.

This will ensure that existing XR applications will keep working and not suddenly break.

In the long run we can remove it.

Reply:
You don't need to add a proxy property, you can handle this in `_set()` and `_get()` directly.

```
#ifndef DISABLE_DEPRECATED
bool Viewport::_set(const StringName &p_name, const Variant &p_value) {
	if (p_name == "use_xr") {
		set_viewport_mode(VIEWPORT_MODE_XR);
		return true;
	}

	return false;
}

bool Viewport::_get(const StringName &p_name, Variant &r_ret) const {
	if (p_name == "use_xr") {
		r_ret = get_viewport_mode() == VIEWPORT_MODE_XR;
		return true;
	}

	return false;
}
#endif
```

Reply:
@JonqsGames or did you mean you already changed this but it's not yet pushed in?

Reply:
c231dbe958b52247afa151d006f552d8247f6913 was supposed to reenable the setter, i choose to not expose the the property to enforce usage of the viewport mode after conversion.

Reply:
@akien-mga ah ok, didn't know that, yeah that's perfect.

Reply:
The compatibility methods suggested by Akien need to be added so this doesn't break backwards compatibility